### ISSUE TITLE
PollingUpdater's onReceive() doing certain operations on main thread causing Application Not Responding error

### ISSUE THREAD

**Issue by balarka** [launchdarkly/android-client-sdk#85]
It appears that PollingUpdate.java onReceive() method is trying to get user from UserManager on main thread which is causing Application Not Responding (ANR) error in our app. 

Version: **com.launchdarkly:launchdarkly-android-client:2.5.0**

Below is the stack trace in Android vitals from Google Play Console.
```
"main" prio=5 tid=1 TimedWaiting
  | group="main" sCount=1 dsCount=0 flags=1 obj=0x73670ca0 self=0x70d32c5a00
  | sysTid=24151 nice=0 cgrp=default sched=0/0 handle=0x70d90009c8
  | state=S schedstat=( 4484132241 6486322535 11275 ) utm=262 stm=186 core=1 HZ=100
  | stack=0x7ff0c29000-0x7ff0c2b000 stackSize=8MB
  | held mutexes=
  at java.lang.Object.wait (Native method)
- waiting on <0x09d42fb6> (a java.lang.Object)
  at java.lang.Thread.parkFor$ (Thread.java:2135)
- locked <0x09d42fb6> (a java.lang.Object)
  at sun.misc.Unsafe.park (Unsafe.java:358)
  at java.util.concurrent.locks.LockSupport.parkNanos (LockSupport.java:230)
  at com.google.a.f.a.a.get (AbstractFuture.java:393)
  at com.google.a.f.a.a$h.get (AbstractFuture.java:86)
  at com.launchdarkly.android.PollingUpdater.onReceive (PollingUpdater.java:34)
  at android.app.ActivityThread.handleReceiver (ActivityThread.java:3390)
  at android.app.ActivityThread.-wrap18 (ActivityThread.java)
  at android.app.ActivityThread$H.handleMessage (ActivityThread.java:1780)
  at android.os.Handler.dispatchMessage (Handler.java:105)
  at android.os.Looper.loop (Looper.java:164)
  at android.app.ActivityThread.main (ActivityThread.java:6938)
  at java.lang.reflect.Method.invoke (Native method)
  at com.android.internal.os.Zygote$MethodAndArgsCaller.run (Zygote.java:327)
  at com.android.internal.os.ZygoteInit.main (ZygoteInit.java:1374)
```
Please fix this asap. Thanks!

**Comment by gwhelanLD** [comment]
Hi @balarka,

Thanks for reaching out. This issue should be fixed as of version 2.5.4 of the Android SDK. There have also been other releases and improvements since then, the current version of this SDK is 2.8.5. See the release notes for more information on the improvements available with an updated version https://github.com/launchdarkly/android-client-sdk/releases.

Thanks,
@gwhelanLD

**Comment by gwhelanLD** [comment]
As this issue should be already fixed, I'm going to go ahead and close it for now. Feel free to reopen this issue if you run into it past version 2.5.4 of the Android SDK.

Thanks,
@gwhelanLD
